integrationtest/showcase/library/src/main/java/org/hibernate/search/integrationtest/showcase/library/repository/indexsearch/IndexSearchDocumentRepositoryImpl.java

```diff
@@ -52,22 +49,18 @@
 			return Optional.empty();
 		}
 
-		// Must use Hibernate ORM types (as opposed to JPA types) to benefit from query.uniqueResult()
-		FullTextSession fullTextSession = Search.getFullTextEntityManager( entityManager ).unwrap( FullTextSession.class );
-
-		org.hibernate.search.mapper.orm.hibernate.FullTextQuery<Book> query =
-				fullTextSession.search( Book.class ).query()
+		SearchQuery<Book> query = Search.getSearchSession( entityManager ).search( Book.class )
 						.asEntity()
 						// onRawField option allows to bypass the bridge in the DSL
 						.predicate( f -> f.match().onRawField( "isbn" ).matching( isbnAsString ) )
-						.build();
+						.toQuery();
 
-		return Optional.ofNullable( query.uniqueResult() );
+		return query.fetchSingleHit();
 	}
 
 	@Override
-	public List<Book> searchByMedium(String terms, BookMedium medium, int offset, int limit) {
-		FullTextQuery<Book> query = Search.getFullTextEntityManager( entityManager ).search( Book.class ).query()
+	public List<Book> searchByMedium(String terms, BookMedium medium, long offset, long limit) {
```

Comment:
Maybe we should change the order of the parameter `offset` and ` limit` into `limit` and `offset`, in order to be more consist with the new Search API.